examples/log.rs

```diff
@@ -1,99 +1,98 @@
-//! Addition example.
 #![cfg_attr(target_arch = "wasm32", no_std)]
 #![cfg_attr(target_arch = "wasm32", no_main)]
 
 extern crate zink;
 
-use zink::Event;
+use zink::{
+    primitives::{Address, U256},
+    Asm, Event,
+};
 
-/// A `Ping` event.
+/// ERC20 standard events
 #[derive(Event)]
-struct Ping;
-
-#[zink::external]
-pub fn log0() {
-    Ping.log0();
+pub enum ERC20Events {
+    /// Emitted when tokens are transferred between addresses
+    /// Parameters: from, to, value
+    Transfer(Address, Address, U256),
+
+    /// Emitted when an address approves another address to spend tokens
+    /// Parameters: owner, spender, value
+    Approval(Address, Address, U256),
 }
 
-#[zink::external]
-pub fn log1() {
-    Ping.log1(b"pong");
+impl ERC20Events {
+    /// Log a transfer event
+    pub fn log_transfer(from: Address, to: Address, value: U256) {
+        match Self::Transfer(from, to, value) {
+            event => event.log0(),
+        }
+    }
+
+    /// Log an approval event
+    pub fn log_approval(owner: Address, spender: Address, value: U256) {
+        match Self::Approval(owner, spender, value) {
+            event => event.log0(),
+        }
+    }
 }
 
-#[zink::external]
-pub fn log2() {
-    Ping.log2(b"pong", b"ping");
+pub mod erc20_events {
+    use super::*;
+
+    /// Logs a transfer event
+    /// Example contract functions demonstrating event logging
+    #[zink::external]
+    pub fn log_transfer(from: Address, to: Address, value: U256) {
+        ERC20Events::log_transfer(from, to, value)
+    }
+
+    /// Logs an approval event
+    /// Example contract functions demonstrating event logging
+    #[zink::external]
+    pub fn log_approval(owner: Address, spender: Address, value: U256) {
+        ERC20Events::log_approval(owner, spender, value)
+    }
+
+    /// Example of logging multiple events in a single transaction
+    /// Example contract functions demonstrating event logging
+    #[zink::external]
+    pub fn log_transfer_and_approval(
+        from: Address,
+        to: Address,
+        spender: Address,
+        value: U256,
+    ) -> Result<(), ()> {
+        // Log transfer first
+        log_transfer(from, to, value);
+        // Then log approval
+        log_approval(from, spender, value);
+        Ok(())
+    }
```

Comment:
The testing interfaces are basically correct! 

for now, let's temporarily forget about the proc-macro, the expanded logic of these methods should be 

```rust
#[zink::external]
pub fn topic0() {
   evm::ffi::log0("MyEvent");
}

#[zink::external]
pub fn topic1(value: U256) {
  ffi::evm::log1("MyEvent", value);
}

// ...
```

I'd suggest to complete this logic first, once it pass, generate them with proc-macro!